src/utils/safe-messages.ts

```diff
@@ -67,3 +72,38 @@ export const supportsEIP1271 = ({ version, fallbackHandler }: SafeInfo): boolean
 
   return gte(version, EIP1271_SUPPORTED_SAFE_VERSION)
 }
+
+export const tryOffChainMsgSigning = async (
+  signer: providers.JsonRpcSigner,
+  safe: SafeInfo,
+  message: SafeMessage['message'],
+): Promise<string> => {
+  const signingMethods = getSupportedSigningMethods(safe.version)
+
+  for await (const [i, signingMethod] of signingMethods.entries()) {
+    try {
+      if (signingMethod === 'eth_signTypedData') {
```

Comment:
Should we also try v3/v4? Or Ledger supports none?

Reply:
Ledger supports none and `ethers` also only knows one. 
If you check the `safe-core-sdk` all of these variants will call the same function in ethers. It only makes a difference when using the `web3` adapter.